src/jllvm/materialization/ByteCodeCompileLayer.cpp

```diff
@@ -1257,11 +1258,21 @@ void codeGenBody(llvm::Function* function, const Code& code, const ClassFile& cl
             // TODO: LCmp
             // TODO: LConst0
             // TODO: LConst1
-            [&](LDC ldc)
+            [&](OneOf<LDC, LDCW, LDC2W> ldc)
             {
+                if (holds_alternative<LDC2W>(operation))
```

Comment:
I don't think you need a separate branch here. You can add `LongInfo` and `DoubleInfo` to `pool` below and simply
add the two lambdas to the `match` below as well.